openedx/core/lib/api/view_utils.py

```diff
@@ -205,3 +206,100 @@ def get_object_or_none(self):
                 # PATCH requests where the object does not exist should still
                 # return a 404 response.
                 raise
+
+
+class LazySequence(Sequence):
+    """
+    This class provides an immutable Sequence interface on top of an existing
+    iterable.
+
+    It is immutable, and accepts an estimated length in order to support __len__
+    without exhausting the underlying sequence
+    """
+    def __init__(self, iterable, est_len=None):  # pylint: disable=super-init-not-called
```

Comment:
Can you add a comment for why init is not called?  Can be done in a follow-up PR if you want.

Reply:
Yes, will do.